website/docs/sdk-reference/dart.md

```diff
@@ -102,6 +104,32 @@ final value = await client.getValue(
 );
 ```
 
+## Anatomy of `getValueDetails()`
+
+`getValueDetails()` is similar to `getValue()` but instead of returning only the evaluated value it gives more detailed information about the evaluation result.
+
+| Parameters     | Description                                                                                                  |
+| -------------- | ------------------------------------------------------------------------------------------------------------ |
+| `key`          | **REQUIRED.** The key of a specific setting or feature flag. Set on *ConfigCat Dashboard* for each setting.                           |
+| `defaultValue` | **REQUIRED.** This value will be returned in case of an error.                                               |
+| `user`         | Optional, *User Object*. Essential when using Targeting. [Read more about Targeting.](advanced/targeting.md) |
+```dart
+final details = await client.getValueDetails(
+    key: 'keyOfMySetting', 
+    defaultValue: false,
+    user: ConfigCatUser(identifier: '#USER-IDENTIFIER#'), // Optional User Object
+);
+
+// Details holds the following information:
+details.value;                              // The evaluated value of the feature flag or setting.
+details.key;                                // The key of the evaluated feature flag or setting.
+details.isDefaultValue;                     // True when the default value passed to getValueDetails() is returned due to an error.
```

Comment:
Is `isDefaultValue` `false` if the default value is returned but there was no error?

Reply:
Technically there's no such case, we always have an error if we decide to return with the default value.

Reply:
I'm asking from the perspective of the user. If I'm the user and I have a few targeting rules to handle exceptional cases, and I have a default value for everything else on the dashboard. Then I feel tempted to check the isDefaultVale to see if the "default value" was returned. 

For example, if I have a flag like this:

canUseConfigCat:
- IF USER's Email IS IN "hacker@qq.com, 123456@qq.com" THEN `FALSE`
- ELSE IF USER's Country IS IN "China" THEN `FALSE`
- ELSE `TRUE`

Here, I think about `TRUE` as the default value.

Reply:
The `isDefaultValue` doesn't indicate that, it's just for signaling that the passed default value was returned. 
The indicator for the case you mentioned (no rules were used for evaluation) is that both `matchedEvaluationRule` and `matchedEvaluationRule` is null.
If I remember correctly, we agreed on not including this case as 'default value' as it's just an internal term used by us.